index.go

```diff
@@ -462,17 +462,22 @@ func readBinary(r io.Reader, data []interface{}) error {
 	return nil
 }
 
-func readIndex(r io.ReadSeeker) (Index, error) {
-	endLastBlock, err := r.Seek(0, io.SeekEnd)
-	if err != nil {
-		return nil, err
+func readIndex(r io.ReadSeeker, o uint64) (Index, error) {
```

Comment:
Maybe it'd be worthy write a comment to know why `readIndex` and `readIndexAt` are different, right now it's confusing.